worker.py

```
@@ -0,0 +1,265 @@
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+)
+
+logger = logging.getLogger(__name__)
+
+# Graceful shutdown flag
+shutdown_flag = False
+
+
+def signal_handler(signum, frame):
```

Ruff (0.14.10)

Unused function argument: `frame`

(ARG001)